code/surface_area/validate_surface_files.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from utils import detect_density
+import nibabel as nib
+import numpy as np
+
+
+def validate_output_file_data(output_metric: Path) -> bool:
+    """
+    Validate the output metric file by checking specific data fields using nibabel.
+    """
+    try:
+        if not output_metric.exists():
+            print("✗ Output file does not exist")
+            return False
+
+        try:
+            gifti_img = nib.load(str(output_metric))
+            print("✓ File loaded successfully with nibabel")
+        except Exception as e:
+            print(f"✗ Failed to load file with nibabel: {e}")
+            return False
+
+        validation_results = {}
+
+        if isinstance(gifti_img, nib.gifti.gifti.GiftiImage):
+            validation_results["Type"] = True
+            print("✓ Type: GIFTI Metric file")
+        else:
+            validation_results["Type"] = False
+            print("✗ Type: Not a GIFTI file")
+
+        num_maps = len(gifti_img.darrays)
+        if num_maps == 1:
+            validation_results["Number of Maps"] = True
+            print(f"✓ Number of Maps: {num_maps}")
+        else:
+            validation_results["Number of Maps"] = False
+            print(f"✗ Number of Maps: Expected 1, got {num_maps}")
+
+        if gifti_img.darrays and len(gifti_img.darrays) > 0:
+            data_array = gifti_img.darrays[0]
+
+            # Get actual data for numeric validation
+            data = data_array.data
+            if data is not None and len(data) > 0:
+                # Number of vertices
+                num_vertices = data.shape[0]
+                density = detect_density(num_vertices)
+                if density in str(output_metric):
+                    validation_results["Number of Vertices"] = True
+                print(f"✓ Number of Vertices: {num_vertices}")
+                print(f"✓ Density: {density}")
```

Comment:
I think this is a good approximation if only validation is being performed and for naming of files. If I am mistaken, I think validation is also done as part of the processing (if the flag is enabled?). If so, I would include the input file here and just validate the number of vertices in the output is equal to the number of vertices in the input.